druid/src/core.rs

```diff
@@ -790,15 +788,17 @@ impl<T: Data, W: Widget<T>> WidgetPod<T, W> {
             }
         };
 
-        // use the substitute event, if one exists
-        let event = substitute_event.as_ref().unwrap_or(event);
+        let mut child_ctx = LifeCycleCtx {
+            command_queue: ctx.command_queue,
+            base_state: &mut self.state,
+            window_id: ctx.window_id,
+        };
 
         if recurse {
-            let mut child_ctx = LifeCycleCtx {
-                command_queue: ctx.command_queue,
-                base_state: &mut self.state,
-                window_id: ctx.window_id,
-            };
+            self.inner.lifecycle(&mut child_ctx, event, data, env);
+        }
+
+        if let Some(event) = substitute_event.as_ref() {
```

Comment:
Given that this is no longer a substitute event, it should be renamed to something like `extra_event`.